### ISSUE TITLE
Ident with SSL doesn't work

### ISSUE THREAD

**Issue by Kegsay** [matrix-org/matrix-appservice-irc#39]
https://matrix.org/jira/browse/BOTS-51

Reconnects forever because we don't get a callback in time.

**Comment by Kegsay** [comment]
Low priority please.

**Comment by Kegsay** [comment]
We can cheekily get around this by using IPv6 \o/

**Comment by warthog9** [comment]
This continues to be an issue, even with IPv6

```
2020-01-10 00:03:05 DEBUG:irc-ident CONNECT 2600:3c02::f03c:91ff:fe59:7d2e 49655
2020-01-10 00:03:05 DEBUG:irc-ident DATA 56950 , 6697

2020-01-10 00:03:05 DEBUG:irc-ident Waiting on Username
2020-01-10 00:03:06 DEBUG:irc-ident Username: |undefined|
2020-01-10 00:03:06 DEBUG:irc-ident No user on port 56950                                                                                                                                                                
2020-01-10 00:03:06 DEBUG:irc-ident 56950,6697:ERROR:NO-USER

2020-01-10 00:03:06 INFO:IrcBridge [8bxldv2d6l800] [M->I] DEAD  (300421ms)                                                                                                                                                                            
2020-01-10 00:03:06 DEBUG:irc-ident Set user <redated username> on port 56950
2020-01-10 00:03:06 DEBUG:irc-ident CLOSE
2020-01-10 00:03:06 ERROR:client-connection <redacted username>@irc.freenode.net: 
```
So the user gets defined after the ident has already responded, and the connection fails.  There is clearly a race of some sort as this will *OCCASIONALLY* complete correctly, though usually by happenstance and usually with the wrong username associated to a connection (though I have seen it).

Switching off of SSL seems to resolve this, but leaves the connections unencrypted (which is not always ideal)
